testing.go

```diff
@@ -263,6 +264,10 @@ func AcceptanceTest(t *testing.T, bkt Bucket) {
 		return nil
 	}))
 	expected = []string{"obj_5.some", "id1/"}
+	if os.Getenv("IS_AZURE_DATA_LAKE_GEN2") == "true" && bkt.Provider() == AZURE {
+		expected = []string{"obj_5.some", "id1/", "id2/"} // Azure Data Lake Gen2 keeps empty dirs.
+	}
+
```

Comment:
If I understand correctly, this ^ is the cause of the problem observed with using the objstore client in Loki where it tries to open a directory and fails.

The loki code is written with the assumption that when all contents of a bucket subpath are removed, the subpath ("directory") is implicitly removed too.

This is the case with a gen1 store, but is _not_ the case for a gen2 store where directories are actual filesystem entities, not just delimiters in the names of objects in a flat namespace. This behaviour difference will persist _irrespective of whether azblob or azdatalake is used as the client_ to communicate with a gen2 store.